src/ai/backend/gateway/groupconfig.py

```diff
@@ -0,0 +1,252 @@
+from .types import CORSOptions, Iterable, WebMiddleware
+from .utils import check_api_params
+
+from ..manager.models import (
+    groups,
+    query_group_dotfiles,
+    verify_dotfile_name,
+    MAXIMUM_DOTFILE_SIZE,
+)
+
+log = BraceStyleAdapter(logging.getLogger('ai.backend.gateway.dotfile'))
+
+
+@server_status_required(READ_ALLOWED)
+@auth_required
+@check_api_params(t.Dict(
+    {
+        tx.AliasedKey(['group', 'groupId', 'group_id']): tx.UUID | t.String,
+        t.Key('domain', default=None): t.String | t.Null,
+        t.Key('data'): t.String(max_length=MAXIMUM_DOTFILE_SIZE),
+        t.Key('path'): t.String,
+        t.Key('permission'): t.Regexp(r'^[0-7]{3}$', re.ASCII),
+    }
+))
+async def create(request: web.Request, params: Any) -> web.Response:
+    log.info('CREATE DOTFILE (group: {0})', params['group'])
+    dbpool = request.app['dbpool']
+    if not request['is_superadmin'] and not request['is_admin']:
+        raise GenericForbidden('Only superadmin or admin can create dotfiles')
+
+    group_id_or_name = params['group']
+    async with dbpool.acquire() as conn, conn.begin():
+        if isinstance(group_id_or_name, str):
+            if params['domain'] is None:
+                raise InvalidAPIParameters('Missing parameter \'domain\'')
+            query = (sa.select([groups.c.id])
+                        .select_from(groups)
+                        .where(groups.c.domain_name == params['domain'])
+                        .where(groups.c.name == group_id_or_name))
+            group_id = await conn.scalar(query)
+        else:
+            group_id = group_id_or_name
+        if group_id_or_name and group_id is None:
+            raise GroupNotFound
+
+        dotfiles, leftover_space = await query_group_dotfiles(conn, group_id)
+        if dotfiles is None:
+            raise GroupNotFound
+        if leftover_space == 0:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+        if len(dotfiles) == 100:
+            raise DotfileCreationFailed('Dotfile creation limit reached')
+        if not verify_dotfile_name(params['path']):
+            raise InvalidAPIParameters('dotfile path is reserved for internal operations.')
+
+        duplicate = [x for x in dotfiles if x['path'] == params['path']]
+        if len(duplicate) > 0:
+            raise DotfileAlreadyExists
+        new_dotfiles = list(dotfiles)
+        new_dotfiles.append({'path': params['path'], 'perm': params['permission'],
+                             'data': params['data']})
+        dotfile_packed = msgpack.packb(new_dotfiles)
+        if len(dotfile_packed) > MAXIMUM_DOTFILE_SIZE:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+
+        query = (groups.update()
+                        .values(dotfiles=dotfile_packed)
+                        .where(groups.c.id == group_id))
+        await conn.execute(query)
+    return web.json_response({})
+
+
+@auth_required
```

Comment:
I think it would be better for group dotfiles not to be exposed to users in other groups as well as the admins of domains where the group does not belong to.